File: src/MentorSync.UI/src/shared/enums/availability.ts
export enum Availability {
    None = 0,
    Morning = 1 << 0, // 1
    Afternoon = 1 << 1, // 2
    Evening = 1 << 2, // 4
    Night = 1 << 3, // 8
}

export const timeOfDayOptions = [
    { value: Availability.Morning, label: "Ранок", desc: "6:00-12:00" },
    { value: Availability.Afternoon, label: "День", desc: "12:00-17:00" },
    { value: Availability.Evening, label: "Вечір", desc: "17:00-22:00" },
    { value: Availability.Night, label: "Ніч", desc: "22:00-6:00" },
];

File: src/Modules/Users/MentorSync.Users/Domain/Enums/Availability.cs
namespace MentorSync.Users.Domain.Enums;

/// <summary>
/// Represents the availability of a mentor during different times of the day.
/// </summary>
[Flags]
public enum Availability
{
	/// <summary>
	/// No availability
	/// </summary>
	None = 0,
	/// <summary>
	/// Available in the morning
	/// </summary>
	Morning = 1 << 0, // 1
	/// <summary>
	/// Available in the afternoon
	/// </summary>
	Afternoon = 1 << 1, // 2
	/// <summary>
	/// Available in the evening
	/// </summary>
	Evening = 1 << 2, // 4
	/// <summary>
	/// Available at night
	/// </summary>
	Night = 1 << 3  // 8
}
